backend/dal/requestDAL_test.go

```diff
@@ -109,3 +109,29 @@ func TestGetLastRequestsBeforeTimestamp(t *testing.T) {
 		})
 	})
 }
+
+func TestInsertRequest(t *testing.T) {
+	// Get connection to test database and cleans it
+	db := tools.GetTestDB()
+	defer tools.CleanDB(db)
+	requestDAL := getRequestDAL(db)
+
+	// Create the fake request
+	var req models.Request
+	req.Name = "Requis Aula Calc"
+	req.Description = "Procuro aula de calculo"
+	req.ExtraInfo = "Informacao X"
+	req.MinPrice = 20
+	req.MaxPrice = 50
+
+	u := tools.CreateFakeUser(t, db, "user", "user@user.com", "1234", "9999-1111")
+	c := tools.CreateFakeCategory(t, db, "Aula Matemática", "Matemática")
+	req.Userid = u.Userid
+	req.Categoryid = c.ID
+
+	// Executes the test query
+	err := requestDAL.InsertRequest(req)
+
+	// Checks the expected results
+	assert.Equal(t, nil, err)
```

Comment:
Wouldn't it be better to test if the request were successfully inserted in the db?
You could call the *GetLastRequests* method, for example, and check if the request was inserted correctly.

Reply:
Check made with another function comparing the result and the to be inserted request.